### PR TITLE
add more rules - Drupal

### LINKED ISSUE #83: Drupal Nginx Conf
Hi @valentinxxx ,

You need to block access to yaml files.

```
location ~ ((^|/)\.|^.*\.yml$|^/sites/.*/private/|^/sites/[^/]+/.*settings.*\.php$) {
        deny all;
        return 404;
}

```
Thanks,
Alexandre

### PR DESCRIPTION
**Pull request by saidatom**
Fixes #83

@MattIPv4 new PR to `master`

### CODE REVIEW

File: src/nginxconfig/generators/conf/drupal.conf.js [hunk 1]
```diff
@@ -18,8 +18,10 @@ export default global => {
     const config = {};
 
     config['# Drupal: deny private files'] = '';
-    config['location ~ ^/sites/.*/private/'] = {
+
```
Comment (MattIPv4) [inline src/nginxconfig/generators/conf/drupal.conf.js:21]:
```suggestion
```
Please remove the extra linebreak :)

File: src/nginxconfig/generators/conf/drupal.conf.js [hunk 2]
```diff
@@ -18,8 +18,10 @@ export default global => {
...
+    config['location ~ ((^|/)\.|^.*\.yml$|^/sites/.*/private/|^/sites/[^/]+/.*settings.*\.php$)'] = {
```
Comment (MattIPv4) [inline src/nginxconfig/generators/conf/drupal.conf.js:22]: In `^/sites/[^/]+/.*settings.*\.php$`, `[^/]+` is used which I assume is meant to handle a single-level-deep subdirectory. This on its own would be fine, however after is `/.*` which will match anything including multi-level-deep subdirectories.

```suggestion
    config['location ~ ((^|/)\.|^.*\.yml$|^/sites/.*/private/|^/sites/[^/]+/[^/]*settings.*\.php$)'] = {
```

Assuming the intention is a single-level-deep subdirectory here, this should fix it.

File: src/nginxconfig/generators/conf/drupal.conf.js [hunk 3]
```diff
@@ -32,6 +34,15 @@ export default global => {
         deny: 'all',
     };
 
+    config['# Allow image styles to be handled by the CMS.'] = '';
```
Comment (MattIPv4) [inline src/nginxconfig/generators/conf/drupal.conf.js:37]:
```suggestion
    config['# Drupal: allow image styles to be handled by the CMS'] = '';
```
Comment consistency in the final config :)

File: src/nginxconfig/generators/conf/drupal.conf.js [hunk 4]
```diff
@@ -32,6 +34,15 @@ export default global => {
...
+    config['location ~ ^/sites/[^/]+/files/styles/'] = {
+        try_files: '$uri @rewrite',
+    };
+
+    config['location @rewrite'] = {
+        rewrite: '^/(.*)$ /index.php?q=$1',
+    };
```
Comment (MattIPv4) [inline src/nginxconfig/generators/conf/drupal.conf.js:44]: Why does this use `rewrite` and not just internally rewrite with `try_files`?

Wouldn't this achieve the same as these two blocks, without sending a rewrite to the client?

```
location ~ ^/sites/[^/]+/files/styles/ {
    try_files $uri /index.php?q=$uri&$args;
}
```

Comment (saidatom) [reply src/nginxconfig/generators/conf/drupal.conf.js]: Yes, you are right.

File: src/nginxconfig/generators/conf/drupal.conf.js [hunk 5]
```diff
@@ -32,6 +33,11 @@ export default global => {
         deny: 'all',
     };
 
+    config['# Drupal: allow image styles to be handled by the CMS'] = '';
+    config['location ~ ^/sites/[^/]+/files/styles/'] = {
+        try_files: '$uri /index.php?q=$uri&$args;',
```
Comment (MattIPv4) [inline src/nginxconfig/generators/conf/drupal.conf.js:38]:
```suggestion
        try_files: '$uri /index.php?q=$uri&$args',
```
Don't need the semi at the end, the conf generator will add that in

**Comment by MattIPv4** [comment]
Hey, @saidatom - Thanks a ton for this sweet PR! 😄

Would you please shoot me an email when you get a chance?
mcowley at digitalocean dot com 🎉

**Pull request merged**